src/validation.h

```diff
@@ -914,6 +917,15 @@ class ChainstateManager
     //! Check to see if caches are out of balance and if so, call
     //! ResizeCoinsCaches() as needed.
     void MaybeRebalanceCaches() EXCLUSIVE_LOCKS_REQUIRED(::cs_main);
+
+    ~ChainstateManager() {
```

Comment:
Did you consider moving this logic to the `BlockManager` dtor instead of here? There's only ever one `BlockManager` instance (inside `ChainstateManager`), so destruction will happen at the same time. All the logic in this function is on `BlockManager` data.

Reply:
Actually, can't you just call `BlockManager::Unload()` in the `BlockManager` dtor?

Reply:
Resolved in 4668ded6d6ea4299d998abbb57543f37519812e2